include/swift/AST/Module.h

```diff
@@ -744,7 +748,7 @@ class FileUnit : public DeclContext {
   // Only allow allocation of FileUnits using the allocator in ASTContext
   // or by doing a placement new.
   void *operator new(size_t Bytes, ASTContext &C,
-                     unsigned Alignment = alignof(FileUnit));
+                     unsigned Alignment = alignof(FileUnit&));
```

Comment:
... your example gives me the errors
>source_file.cpp(12): error C2027: use of undefined type 'Foo'
>source_file.cpp(10): note: see declaration of 'Foo'
>source_file.cpp(12): error C2338: aligning reference rather than Foo